### DISCUSSION TITLE
StoreFront - TypeError: ReactDOMServer.renderToReadableStream is not a function

### DISCUSSION THREAD

**Discussion by giannineves** [saleor/saleor#9756] [General]
My docker-compose up command works fine. When I try to open the store front I receive the error below:

INTERNAL SERVER ERROR.

**TypeError: ReactDOMServer.renderToReadableStream is not a function**

The dashboard works fine. No errors

**Comment by karolkielecki** [comment]
Hi @giannineves, please provide more info. Is that saleor-platform?

**Reply by dr-rompecabezas** [reply]
Hi @karolkielecki,

I have the same issue as @giannineves. Like @giannineves, I changed the dependencies in package.json from React 18 to React 17 due to a conflict with another package that was preventing Docker from completing the initial installation. See discussion #9736.

After manually changing the dependencies (there was another one related to GraphQL, Docker successfully completed the build. However, now I see that it appears React 18 was required after all (see references to experimental features in logs below).

Here are the logs when starting the react-storefront-1 container:

```
react-storefront_1  | 
react-storefront_1  | > react-storefront@0.1.0 dev /app
react-storefront_1  | > next dev --port 3001
react-storefront_1  | 
react-storefront_1  | ready - started server on 0.0.0.0:3001, url: http://localhost:3001
react-storefront_1  | info  - Loaded env from /app/.env
react-storefront_1  | warn  - You have enabled experimental feature(s).
react-storefront_1  | warn  - Experimental features are not covered by semver, and may cause unexpected or broken application behavior. Use them at your own risk.
react-storefront_1  | 
react-storefront_1  | warn  - SWC minify release candidate enabled. https://nextjs.org/docs/messages/swc-minify-enabled

react-storefront_1  | warn  - You have to use React 18 to use `experimental.reactRoot`.

react-storefront_1  | event - compiled client and server successfully in 6.4s (493 modules)
```

And here are the logs when hitting the `http://localhost:3001/` endpoint, which returns a 500 HTTP status - internal server error.
```
react-storefront_1  | warn  - using beta Middleware (not covered by semver) - https://nextjs.org/docs/messages/beta-middleware
react-storefront_1  | wait  - compiling /_middleware...
react-storefront_1  | event - compiled client and server successfully in 1168 ms (544 modules)
react-storefront_1  | wait  - compiling /[channel]/[locale] (client and server)...
react-storefront_1  | event - compiled client and server successfully in 3.9s (978 modules)
api_1               | [16/May/2022 17:16:02] "POST /graphql/ HTTP/1.0" 200 836
react-storefront_1  | wait  - compiling /_error (client and server)...
react-storefront_1  | event - compiled client and server successfully in 495 ms (979 modules)
react-storefront_1  | TypeError: ReactDOMServer.renderToReadableStream is not a function
react-storefront_1  |     at Object.renderToInitialStream (/app/node_modules/.pnpm/next@12.1.4_sfoxds7t5ydpegc3knd667wn6m/node_modules/next/dist/server/node-web-streams-helper.js:204:33)
react-storefront_1  |     at renderDocument (/app/node_modules/.pnpm/next@12.1.4_sfoxds7t5ydpegc3knd667wn6m/node_modules/next/dist/server/render.js:849:61)
react-storefront_1  |     at Object.renderToHTML (/app/node_modules/.pnpm/next@12.1.4_sfoxds7t5ydpegc3knd667wn6m/node_modules/next/dist/server/render.js:906:34)
react-storefront_1  |     at async doRender (/app/node_modules/.pnpm/next@12.1.4_sfoxds7t5ydpegc3knd667wn6m/node_modules/next/dist/server/base-server.js:878:38)
react-storefront_1  |     at async cacheEntry.responseCache.get.isManualRevalidate.isManualRevalidate (/app/node_modules/.pnpm/next@12.1.4_sfoxds7t5ydpegc3knd667wn6m/node_modules/next/dist/server/base-server.js:977:28)
react-storefront_1  |     at async /app/node_modules/.pnpm/next@12.1.4_sfoxds7t5ydpegc3knd667wn6m/node_modules/next/dist/server/response-cache.js:72:36
react-storefront_1  | warn  - You have to use React 18 to use `experimental.reactRoot`.
react-storefront_1  | error - TypeError: ReactDOMServer.renderToReadableStream is not a function
```
What do you suggest? Thanks for your help.

**Reply by giannineves** [reply]
Hi @karolkielecki and @dr-rompecabezas  ,

I found the problem. When you set experimental/reactRoot as true, you have this error. So, to fix the issue, at n**ext.config.js in REACT-STOREFRONT,** you should update the line below to FALSE:

--issue
experimental: {
    reactRoot: true,
  },

--correction
experimental: {
    reactRoot: false,
  },

**Reply by dr-rompecabezas** [reply]
Nice find @giannineves. Thank you!

**Reply by karolkielecki** [reply]
Btw, it seems you're using an older version of React. You received a few warn `warn  - You have to use React 18 to use `experimental.reactRoot`.`

**Reply by giannineves** [reply]
Hi @karolkielecki,

Yes, you're right, the version I am using is ^17.0.2. The problem was when using react and react-dom 18, I hit even more compatibilities issues. If using experimental.reactRoot is important to the project, let me know and I will try to make things work with 18.0.0 version, but will take me a little more time.

Thanks @karolkielecki
